examples_generator/libgen.py

```diff
@@ -1480,37 +1480,42 @@ def __init__(self, infra):
             infra.lengths[oget_track(otrack)] for otrack in range(2 * infra.nb_tracks)
         ]
 
-    def route_path(self, departure_utrack, arrival_utrack):
-        s1, s2 = uget_begin(departure_utrack), uget_end(departure_utrack)
-        t1, t2 = uget_begin(arrival_utrack), uget_end(arrival_utrack)
-        parent = [None for u in self.neighboors]
-        q = [(0, s1, -1), (0, s2, -1)]
-        while q != [] and parent[t1] == None and parent[t2] == None:
-            d, u, p = heappop(q)
-            if parent[u] == None:
-                parent[u] = p
-                for v in self.neighboors[u]:
-                    heappush(q, (d + self.weight[v], v, u))
-        path = []
-        u = t1 if parent[t1] != None else t2
-        while u != -1:
-            path.append(u)
-            u = parent[u]
-        path.reverse()
-        path = [oget_other_side(path[0])] + path + [oget_other_side(path[-1])]
-        return [oname_route_between(path[i], path[i + 1]) for i in range(len(path) - 1)]
-
-    def add_schedule(self, departure_time, departure_utrack, arrival_utrack):
-        assert departure_utrack != arrival_utrack
-        path = self.route_path(departure_utrack, arrival_utrack)
+    def route_path(self, track_list):
+        final_path=[]
+        for i in range(len(track_list)-1):
+            s1, s2 = uget_begin(track_list[i]), uget_end(track_list[i])
+            t1, t2 = uget_begin(track_list[i+1]), uget_end(track_list[i+1])
+            parent = [None for u in self.neighboors]
+            q = [(0, s1, -1), (0, s2, -1)]
+            while q != [] and parent[t1] == None and parent[t2] == None:
+                d, u, p = heappop(q)
+                if parent[u] == None:
+                    parent[u] = p
+                    for v in self.neighboors[u]:
+                        heappush(q, (d + self.weight[v], v, u))
+            path = []
+            u = t1 if parent[t1] != None else t2
+            while u != -1:
+                path.append(u)
+                u = parent[u]
+            path.reverse()
+            path = [oget_other_side(path[0])] + path + [oget_other_side(path[-1])]
+            if final_path!=[]:
+                final_path.pop()
+            final_path=final_path+[oname_route_between(path[i], path[i + 1]) for i in range(len(path) - 1)]
+        return final_path
+
+    def add_schedule(self, departure_time, track_list):
```

Comment:
You can have a better API with a variadic function like this:
```suggestion
    def add_schedule(self, departure_time, *track_list):
```

You can use `track_list` will be a list, but the call to `add_schedule()` will change like that:

```python
# Previously
sim.add_schedule(8000, [1, 6, 7])
# With new API
sim.add_schedule(8000, 1, 6, 7)
```